internal/validator/handler.go

```diff
@@ -42,18 +44,45 @@ type handler struct {
 	producer *kafka.Producer
 	schema   *jsonschema.Schema
 	errors   chan error
+	msgChan  chan *kafka.Message
+	vChan    chan messageInfo
 }
 
 func (this *handler) onMessage(ctx context.Context, msg *kafka.Message) {
-	request := &messageModel.IngressValidationRequest{}
-	err := json.Unmarshal(msg.Value, request)
+	wg.Add(1)
 
-	if err != nil {
-		instrumentation.UnmarshallingError(ctx, err)
-		return
-	}
+	go func() {
+		defer wg.Done()
+
+		for {
+			if len(this.msgChan) >= cap(this.msgChan) {
+				continue
+			}
+			this.msgChan <- msg
+			break
+		}
+	}()
+	wg.Wait()
+}
+
+func (this *handler) fetchMessage(
```

Comment:
In general it's better to keep the actual logic of a function (i.e. the sequence of steps performed for one request) separate from the auxiliary code (loop, reading channels, writing channels) which can reside in a separate function. That way it's easier to test the actual logic in separation